lib/xenops_server.ml

```diff
@@ -2338,6 +2341,17 @@ module VM = struct
          let vm_id = VGPU_DB.vm_of vgpu_id in
          match context.transferred_fd with
          | Some transferred_fd ->
+
+           (* prevent vgpu-migration from pre-Jura to Jura and later *)
+           let _ = try List.assoc key_query_of_sender cookies
+             with Not_found ->
+                  (* only Jura and later hosts send this key; fail the migration from pre-Jura hosts *)
+                  let msg = Printf.sprintf "VM.migrate: sending host is too old: no %s" key_query_of_sender in
```

Comment:
`VM.migrate: XenServer version of sending host does not support vGPU migration as it did not send %s cookie`

Reply:
The error here, using version, would be more meaningful:
```
VM.migrate: Unsupported protocol version for vGPU migration. Expected version = 2, but version not present
```

Reply:
The alternative to a version number would be a list/set of supported features as the value. This would have the advantage of being more explicit and we could mix and match features. We could also use an integer as a bit set with bits assigned to features.